src/dotnet/Program.cs

```diff
@@ -113,6 +113,16 @@ private static int ProcessArgs(string[] args)
 
             var appArgs = (lastArg + 1) >= args.Length ? Enumerable.Empty<string>() : args.Skip(lastArg + 1).ToArray();
 
+            if (verbose.HasValue)
+            {
+                Environment.SetEnvironmentVariable(CommandContext.Variables.Verbose, verbose.ToString());
+            }
+            //This process should inherit AnsiPassThrough or use the default of false.
+            CommandContext.InitializeAnsiPassThroughFromEnvironment();
```

Comment:
Yeah, it does not seem a bit hacky.

I think it would not be too hard to make all external external commands. Since all the commands use [Command.CreateDotnet](https://github.com/dotnet/cli/blob/65c026afbf2f7dab3ce9fd1ef480ae6233d5fb54/src/Microsoft.DotNet.Cli.Utils/Command.cs#L43), I could change that to apply the ANSI forwarding.